cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -1187,24 +1201,84 @@ var (
 			if err != nil {
 				return err
 			}
+
+			// Determine which devices to check for unclaim support.
 			for _, dev := range devices.GetEndDevices() {
-				// Check if the device is in the configured cluster.
-				nsMismatch, asMismatch, jsMismatch := compareServerAddressesEndDevice(dev, config)
-				if nsMismatch || asMismatch || jsMismatch {
+				nsMismatch, asMismatch, _ := compareServerAddressesEndDevice(dev, config)
+				if nsMismatch || asMismatch {
 					return errAddressMismatchEndDevice.New()
 				}
 				del = append(del, dev.GetIds().GetDeviceId())
+				if dev.JoinServerAddress == "" && dev.GetIds().GetJoinEui() != nil {
+					unclaim = append(unclaim, dev.GetIds().GetDeviceId())
+					// This request will have duplicates but that's allowed by the DCS.
+					infoByJoinEUIs = append(infoByJoinEUIs, &ttnpb.GetInfoByJoinEUIRequest{
+						JoinEui: dev.GetIds().GetJoinEui(),
+					})
+				} else {
+					jsDevs = append(jsDevs, dev)
+				}
+			}
+
+			dcs, err := api.Dial(ctx, config.DeviceClaimingServerGRPCAddress)
+			if err != nil {
+				return err
+			}
+			dcsBatchClient := ttnpb.NewEndDeviceBatchClaimingServerClient(dcs)
+
+			claimInfosResp, err := dcsBatchClient.GetInfoByJoinEUIs(
+				ctx,
+				&ttnpb.GetInfoByJoinEUIsRequest{
+					Requests: infoByJoinEUIs,
+				},
+			)
+			if err != nil {
+				return err
+			}
+			for _, claimInfo := range claimInfosResp.Infos {
+				if !claimInfo.SupportsClaiming {
+					// These devices are selected since they are not registered with the cluster Join Server.
+					// If these EUIs are not claimable, then there's something wrong with the registration.
```

Comment:
Or there's something wrong with the JS configuration.

It could be that the JS has been removed from DCS, so it doesn't support claiming (anymore), but the end device is still there with an empty JS address.

If you return the error here, the user has no way to recover for this.